src/useMutationObserver.ts

```diff
@@ -0,0 +1,68 @@
+import useCustomEffect, { IsEqual } from './useCustomEffect'
+import isEqual from 'lodash/isEqual'
+import useImmediateUpdateEffect from './useImmediateUpdateEffect'
+import useMountEffect from './useMountEffect'
+import useEventCallback from './useEventCallback'
+import { useRef } from 'react'
+
+type Deps = [Element | null | undefined, MutationObserverInit]
+
+const isDepsEqual: IsEqual<Deps> = (prev, next) =>
+  prev[0] === next[0] && isEqual(prev[1], next[1])
```

Comment:
you probably want "shallow equal" for the config.

actually we probably want a helper that does "shallow equal for each member of deps array"